src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c

```diff
@@ -324,11 +324,15 @@ static inline void read_ert_stat(struct kds_command *xcmd)
 	mutex_lock(&kds->scu_mgmt.lock);
 	/* Skip header and FPGA CU stats. off_idx points to PS kernel stats */
 	off_idx = 4 + num_cu;
-	for (i = 0; i < num_scu; i++)
-		kds->scu_mgmt.usage[i] = ecmd->data[off_idx + i];
+	for (i = 0; i < num_scu; i++) {
+		kds->scu_mgmt.usage[i] = ecmd->data[off_idx++];
+		kds->scu_mgmt.sc_usages[i].succ_cnt = ecmd->data[off_idx++];
+                kds->scu_mgmt.sc_usages[i].err_cnt = ecmd->data[off_idx++];
+                kds->scu_mgmt.sc_usages[i].crsh_cnt = ecmd->data[off_idx++];
+        }
 
 	/* off_idx points to PS kernel status */
-	off_idx += num_scu + num_cu;
+	off_idx += num_cu;
```

Comment:
This calculation doesnt seem to be correct. Please check

Reply:
Actually there the communication between Xocl and Zocl through packet payload. 
The payload structure ads follows for this structure.
+------------------------+
|     custat version          |  4 (32bit)
|     git version               |
|     # CQ slots               |
|     # CUs                      |
+------------------------+
| individual CU stat        |  num_cus
|        Total usages          |
+------------------------+
| individual SCU stat       | num_scu
|        Total usages          |
|		 Succ Count    |    
|		 Error Count    |
|		 Crash Count  |
+------------------------+
| individual CU status     | num_cus
|                                      |
+------------------------+
| individual SCU status   | num_scu
|                                      |
+------------------------+
| SK Memory status       | 
|     Hbo count               |
|     Map count               |
|     Unmap count           |
|     Free count               |
+------------------------+
|  slot status                   |  num_slots
+------------------------+

Reply:
As per the offline discussion with you this calculation is fine.